app/uk/gov/hmrc/gform/formtemplate/FormTemplateValidator.scala

```diff
@@ -149,17 +149,19 @@ object FormTemplateValidator {
     }
   }
 
-  def validateEnrolmentIdentifier(formTemplate: FormTemplate): ValidationResult = {
-    val userContextComponentType = formTemplate.expandFormTemplate.allFCs.collect {
-      case expr @ HasExpr(SingleExpr(UserCtx(EnrolledIdentifier))) => expr
+  val userContextComponentType: List[FormComponent] => List[FormComponent] =
+    enrolledIdentifierComponents =>
+      enrolledIdentifierComponents.collect {
+        case expr @ HasExpr(SingleExpr(UserCtx(EnrolledIdentifier))) => expr
     }
 
-    formTemplate.authConfig match {
-      case HmrcSimpleModule | HmrcAgentModule(_) if !userContextComponentType.isEmpty =>
-        Invalid("Invalid auth type for component type.")
-      case _ => Valid
-    }
-  }
+  def validateEnrolmentIdentifier(formTemplate: FormTemplate): ValidationResult =
+    if (userContextComponentType(formTemplate.expandFormTemplate.allFCs).nonEmpty) {
```

Comment:
More clear way would be to use `formTemplate.expandFormTemplate.allFCs.exists(...)`. There is no need to traverse whole list.